packages/ui/src/components/ListV2/Row.tsx

```diff
@@ -0,0 +1,197 @@
+import styled from '@emotion/styled'
+import type { ReactNode } from 'react'
+import { useEffect } from 'react'
+import { SENTIMENTS } from '../../theme'
+import { Checkbox } from '../Checkbox'
+import { Tooltip } from '../Tooltip'
+import { Cell } from './Cell'
+import { useListContext } from './ListContext'
+
+const ExpandableWrapper = styled.div`
+  grid-column: 1 / -1;
+  grid-column-start: 1;
+  border-top: 1px solid ${({ theme }) => theme.colors.neutral.borderWeak};
+  margin: 0 -${({ theme }) => theme.space['2']};
+  padding: ${({ theme }) => theme.space['2']};
+`
+
+export const StyledRow = styled.div`
+  // List itself also apply style about common templating between HeaderRow and other Rows
+
+  position: relative;
+  border: 1px solid ${({ theme }) => theme.colors.neutral.borderWeak};
+  border-radius: ${({ theme }) => theme.radii.default};
+  transition: box-shadow 200ms ease, border-color 200ms ease;
+  box-shadow: none;
+  background-color: ${({ theme }) => theme.colors.neutral.background};
+  cursor: auto;
+  font-size: ${({ theme }) => theme.typography.bodySmall.fontSize};
+  column-gap: ${({ theme }) => theme.space['2']};
+  padding: 0 ${({ theme }) => theme.space['2']};
+
+  &[role='button row'] {
+    cursor: pointer;
+  }
+
+  ${({ theme }) =>
+    SENTIMENTS.map(
+      sentiment => `
+    &[data-sentiment="${sentiment}"] {
+      color: ${theme.colors[sentiment].text};
+      border-color: ${theme.colors[sentiment].border};
+      background-color: ${theme.colors[sentiment].background};
+    }
+
+    &[data-sentiment="${sentiment}"] [data-expandable-content] {
+      border-color: ${theme.colors[sentiment].border};
+    }
+    `,
+    ).join(' ')}
+
+  &[data-sentiment="neutral"]:hover {
+    border-color: ${({ theme }) => theme.colors.primary.border};
+    box-shadow: ${({ theme }) => theme.shadows.hoverPrimary};
+  }
+
+  &[data-highlight='true'] {
+    border-color: ${({ theme }) => theme.colors.primary.border};
+    box-shadow: ${({ theme }) => theme.shadows.hoverPrimary};
+  }
+
+  &[aria-disabled='true'] {
+    border: 1px solid ${({ theme }) => theme.colors.neutral.borderDisabled};
+    background-color: ${({ theme }) => theme.colors.neutral.backgroundDisabled};
+    color: ${({ theme }) => theme.colors.neutral.textDisabled};
+    pointer-events: none;
+  }
+
+  & [data-visibility='hover'] {
+    opacity: 0;
+  }
+  &:hover [data-visibility='hover'] {
+    opacity: 1;
+  }
+`
+
+const StyledCheckboxContainer = styled.div`
+  display: flex;
+`
+
+const StyledCheckbox = styled(Checkbox)`
+  &[aria-disabled='true'] {
+    pointer-events: none;
+  }
+`
+
+type RowProps = {
+  children: ReactNode
+  id: string
+  expandable?: ReactNode
+  isSelectDisabled?: boolean
+  selectTooltip?: string
+  isDisabled?: boolean
+  sentiment?: (typeof SENTIMENTS)[number]
+}
+
+export const Row = ({
```

Comment:
I think we should forward ref at least in the component and maybe in others.
I have already have an use case where I needed to attach the ref to manage drag & drop.

Reply:
Good idea, i add it on List, List.Row & List.Cell, make less sense on List.SelectBar